File: foyer-management-frontend/src/app/models/ressource.ts
import { Profil } from "./profil";
import { Application } from "./Application";
export interface Ressource {
  code: string;
  libelle: string;
  typeRessource: string;
  statut: boolean;
    tempsEstimeJours?: number;  // <- Ajoute ce champ optionnel

  application?: Application;
  profils?: Profil[];
}

File: habilitationbna/src/main/java/com/bna/habilitationbna/model/Ressource.java
package com.bna.habilitationbna.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.Entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.Set;

@Entity
@Getter
@Setter
public class Ressource {
        @Id
        @Column( nullable = false, unique = true)
        private String code;
        private String ipfsUrl;
        @Column(nullable = false)
        private String libelle;

        @Column
        private String typeRessource;
        @Column
        private Double tempsEstimeJours;


        @Column()
        private boolean statut;
        @ManyToOne
        @JoinColumn()
        private Application application;

        @JsonIgnoreProperties("ressources") // Solution alternative

// relation inverse (facultative si bidirectionnelle)
        @ManyToMany(mappedBy = "ressources")
        private Set<Profil> profils;
}
